pdns/opensslsigners.cc

```diff
@@ -353,18 +354,17 @@ void OpenSSLRSADNSCryptoKeyEngine::fromISCMap(DNSKEYRecordContent& drc, std::map
     throw runtime_error(getName()+" tried to feed an algorithm "+std::to_string(drc.d_algorithm)+" to a "+std::to_string(d_algorithm)+" key");
   }
 
-  int ret = RSA_check_key(key);
-  if (ret != 1) {
-    RSA_free(key);
-    throw runtime_error(getName()+" invalid public key");
-  }
-
   if (d_key)
     RSA_free(d_key);
 
   d_key = key;
 }
 
+bool OpenSSLRSADNSCryptoKeyEngine::checkKeys() const
```

Comment:
Could be, it checks a key-pair but we mostly refer to it as key, so you are probably right, I'll change that along with the hsm stuff!